Automattic/jetpack
# Add/subscriber post auth tracking

Fixes 157-gh-automattic/gold

## Proposed changes:

* Changes the logic for `user_has_access` slightly to check for all conditions and then report the status via an action.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






1. Apply D122899-code to your sandbox
2. Make sure store sandbox is enabled and forward your memberships testing site, public api and subscribe.wordpress.com to your sandbox.
3. SSH into your sandbox and open `tail -f /tmp/php-errors`
4. View a protected post as a subscriber (paid or non-paid).

You should see the `wpcom_memberships_successful_auth_for_protected_post` tracks event logged to your php-errors output.
